Profiler

noamtamim opened the pull request.

noamtamim committed:
Initial work on profiler

noamtamim committed:
Merge remote-tracking branch 'origin/dev' into profiler

# Conflicts:
#	playkit/src/main/java/com/kaltura/playkit/player/ExoPlayerWrapper.java
#	playkit/src/main/java/com/kaltura/playkit/player/MediaPlayerWrapper.java
#	playkit/src/main/java/com/kaltura/playkit/player/PlayerController.java
#	playkit/src/main/java/com/kaltura/playkit/player/PlayerEngine.java

noamtamim committed:
More events

noamtamim committed:
More profiling

noamtamim committed:
Merge remote-tracking branch 'origin/dev' into profiler

noamtamim committed:
Profiler: use the new `AnalyticsListener`

noamtamim committed:
Update gradle

noamtamim committed:
Fix how errors how logged

noamtamim committed:
Merge remote-tracking branch 'origin/dev' into profiler

noamtamim committed:
Add sessionId and Android API level to log

noamtamim committed:
Rename listener getter

noamtamim committed:
Add Utils.executeGet()

noamtamim committed:
Send logs to server

No longer saving in files.

noamtamim committed:
Cache the config file

noamtamim committed:
Fix default post url and other fixes

noamtamim committed:
Log more device info

noamtamim committed:
PKDeviceCaps takes chipset from MediaSupport

noamtamim committed:
Merge remote-tracking branch 'origin/dev' into profiler

noamtamim committed:
Some fixes and devMode

devMode: save logs on device

noamtamim committed:
Update Profiler config url

noamtamim committed:
Workaround for the missing onViewportSizeChange()

noamtamim committed:
Fix POST

noamtamim committed:
Use float seconds instead of ms everywhere

noamtamim committed:
Null-related fixes

noamtamim committed:
Merge remote-tracking branch 'origin/dev' into profiler

# Conflicts:
#	playkit/src/main/java/com/kaltura/playkit/player/ExoPlayerWrapper.java

noamtamim committed:
Merge remote-tracking branch 'origin/dev' into profiler

# Conflicts:
#	playkit/src/main/java/com/kaltura/playkit/player/PlayerController.java

noamtamim committed:
Profiler: add duration, remove "released"

noamtamim committed:
Profiler fixes

noamtamim committed:
Merge remote-tracking branch 'origin/dev' into profiler

noamtamim committed:
Simplify Profiler lifecycle logic

noamtamim committed:
Add tracks info. WIP.

noamtamim committed:
Profiler: add width and height to track info

noamtamim committed:
Fix cross-protocol redirects

noamtamim committed:
Profiler: cleanup logLoadingEvent

noamtamim committed:
Profiler: show mime type in tracks format

noamtamim committed:
Remove commented-out code

noamtamim committed:
Add notice on the ViewportSize workaround

noamtamim committed:
Make Profiler an interface

noamtamim committed:
Change Profiler to abstract class

Move some functionality up.

noamtamim committed:
Remove `public` from many members

noamtamim committed:
Profiler: rearrange code

noamtamim committed:
Set exoplayer version in a variable

noamtamim committed:
Profiler: add server info (ip)

noamtamim committed:
Profiler: full URLs in load events

noamtamim committed:
Profiler: improve tracks display

noamtamim committed:
Get the profiler ready

noamtamim committed:
Merge remote-tracking branch 'origin/dev' into profiler

# Conflicts:
#	playkit/src/main/java/com/kaltura/playkit/player/PlayerController.java

noamtamim committed:
Default to enabled -- for now

noamtamim committed:
Merge remote-tracking branch 'origin/dev' into profiler

# Conflicts:
#	playkit/build.gradle
#	playkit/src/main/java/com/kaltura/playkit/player/ExoPlayerWrapper.java

noamtamim committed:
Log server info when loading from a new server

noamtamim committed:
Merge remote-tracking branch 'origin/dev' into profiler

# Conflicts:
#	playkit/src/main/java/com/kaltura/playkit/player/ExoPlayerWrapper.java

noamtamim committed:
Merge remote-tracking branch 'origin/dev' into profiler

# Conflicts:
#	playkit/src/main/java/com/kaltura/playkit/player/ExoPlayerWrapper.java

noamtamim committed:
Change in how the DataSourceFactory is built

noamtamim committed:
Profiler: log okhttp events

noamtamim committed:
Profiler: log the correct hostname after redirect

noamtamim committed:
Merge remote-tracking branch 'origin/dev' into profiler

# Conflicts:
#	playkit/src/main/java/com/kaltura/playkit/player/ExoPlayerWrapper.java

noamtamim committed:
Profiler: log player load control params

noamtamim committed:
[Profiler] Remove referer

noamtamim committed:
Profiler: support more data types

noamtamim committed:
Add toString() to PKMediaSource and PKDrmParams

noamtamim committed:
Merge remote-tracking branch 'origin/dev' into profiler

# Conflicts:
#	playkit/src/main/java/com/kaltura/playkit/player/ExoPlayerWrapper.java

noamtamim committed:
Instead of empty implementations use default

noamtamim committed:
Use a shared conn. pool that the app can warm up

noamtamim committed:
Profiler: add experiments

noamtamim committed:
Merge remote-tracking branch 'origin/dev' into profiler

# Conflicts:
#	playkit/src/main/java/com/kaltura/playkit/player/ExoPlayerWrapper.java

noamtamim committed:
Profiler: black line between chunks

noamtamim committed:
Connection pool warmup: all calls concurrently

But wait for all.

kaltura-hooks commented:
Hi @noamtamim,
Thank you for contributing this pull request!
Please sign the Kaltura CLA so we can review and merge your contribution.
Learn more at http://bit.ly/KalturaContrib

noamtamim committed:
Fix the connection pool

Derive ExoPlayer's http client from the one used by the connection pool manager to make sure it uses the same SSLSocketFactory.

noamtamim committed:
No need to call each warmup url twice

noamtamim committed:
Actually the second time does improve things

noamtamim committed:
Use OKHttp 3.12.1

noamtamim committed:
Don't follow redirects when warming up

noamtamim committed:
Merge remote-tracking branch 'origin/dev' into profiler

kaltura-hooks commented:
Hi @noamtamim,
Thank you for contributing this pull request!
Please sign the Kaltura CLA so we can review and merge your contribution.
Learn more at http://bit.ly/KalturaContrib

noamtamim committed:
Move connection pool manager to .player package

noamtamim committed:
Merge remote-tracking branch 'origin/dev' into profiler

# Conflicts:
#	playkit/src/main/java/com/kaltura/playkit/player/ExoPlayerWrapper.java

noamtamim committed:
Separate the profiler, step 1

noamtamim committed:
Profiler: cleanups

noamtamim committed:
Delete PKConnectionPoolManager (merge leftover)

noamtamim committed:
Change the way listeners are passed

noamtamim committed:
Break the build to test jitpack

noamtamim committed:
Fix the build

It was broken on purpose
to test JitPack

noamtamim committed:
Move profiler implementation to separate module

Some classes/methods are now public to allow it.

noamtamim committed:
Remove the NULL profiler

Now the main class is the null profiler.

noamtamim committed:
Let the app initialize the profiler

It's safer and doesn't require reflection.

noamtamim committed:
Undo changes that don't really belong here

noamtamim committed:
More profiler cleanups

noamtamim committed:
Merge remote-tracking branch 'origin/dev' into profiler

# Conflicts:
#	playkit/src/main/java/com/kaltura/playkit/player/ExoPlayerWrapper.java

noamtamim committed:
Add the profiler as a module

noamtamim committed:
Merge remote-tracking branch 'origin/dev' into profiler

noamtamim committed:
Annotate ExoPlayerWrapper.PlayerSettings NonNull

noamtamim committed:
Comment

noamtamim committed:
Add profiler to settings.gradle

noamtamim committed:
Safety

noamtamim committed:
Merge remote-tracking branch 'origin/dev' into profiler

noamtamim committed:
Publish profiler

noamtamim committed:
Fix gradle publish

noamtamim committed:
Structure change

noamtamim committed:
Rearrange code

noamtamim committed:
The actual factory is a callable

noamtamim committed:
Minor

noamtamim committed:
Publish the profiler as 'playkit-profiler'

noamtamim committed:
PlayKitProfiler is not the actual profiler imp

noamtamim committed:
NULL handling

noamtamim committed:
Experiments cleanup + log package name

noamtamim committed:
Add onApplicationPaused and -Resumed to profiler

noamtamim committed:
PlayKit fixes

noamtamim committed:
Start checking network type

noamtamim committed:
comments from code review

noamtamim committed:
Merge remote-tracking branch 'origin/dev' into profiler

# Conflicts:
#	playkit/src/main/java/com/kaltura/playkit/Utils.java
#	playkit/src/main/java/com/kaltura/playkit/player/Profiler.java

noamtamim committed:
Fix some Codacy issues

noamtamim committed:
Explain the random/percentage logic

noamtamim committed:
Comments, explanations, etc

noamtamim committed:
Magic number order

giladna committed:
Merge branch 'dev' into profiler

# Conflicts:
#	.travis.yml

giladna committed:
move profiler module into playkit

giladna committed:
Update .travis.yml

giladna committed:
Delete travis-push-to-bintray.sh

build upload done via jenkins
